device/globals/global_vm.gd

```diff
@@ -146,24 +146,36 @@ func update_camera(time):
 
 	var cpos = camera.get_position()
 
+	# The camera position is set to target when it's about to overstep it,
+	# or when it's moved there instantly.
+	# Compare the camera and target position until then
 	if cpos != pos:
-		#return
+		var v = pos - cpos  # vector to move along
+		var step = cam_speed * time  # pixel size of step to move
 
-		var dif = pos - cpos
-		var dist = cam_speed * time
-		if dist > dif.length() || cam_speed == 0:
+		# This is where we may overstep or move instantly
+		if step > v.length() || cam_speed == 0:
 			camera.set_position(pos)
-			#return
 		else:
-			camera.set_position(cpos + dif.normalized() * dist)
-			pos = cpos + dif.normalized() * dist
+			pos = cpos + v.normalized() * step
+			camera.set_position(pos)
 
 	var half = game_size / 2
-	pos = _adjust_camera(pos)
+
+	var clamp_data = []
+	var camera_clamped
+
+	clamp_data = _clamp_camera(pos, half)
+	camera_clamped = clamp_data[0]
+	pos = clamp_data[1]
```

Comment:
I think declaration and assignment should be on the same line in this case, since it's immediately clear that `clamp_data` is an array, since values are fetched by index on the next line, ie.:

    var clamp_data = _clamp_camera(pos, half)
    var camera_clamped = clamp_data[0]
    var pos = clamp_data[1]

Reply:
Actually `var pos` was not on that line, I restored it to where it originally was, along with the other `var pos`, so that should be all fine now.

Reply:
Thanks, @mjtorn. 🙂 Merging this now.